fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/BaseKeyExpression.java

```diff
@@ -39,17 +39,21 @@
     @Override
     @Nonnull
     public final KeyExpression getSubKey(int start, int end) {
+        final int columnSize = getColumnSize();
+        if (start < 0 || end > columnSize || start > end) {
+            throw new IllegalSubKeyException(start, end, columnSize);
+        }
         if (start == end) {
             return EmptyKeyExpression.EMPTY;
-        } else if ((start == 0) && (end == getColumnSize())) {
+        } else if ((start == 0) && (end == columnSize)) {
             return this;
         }
         return getSubKeyImpl(start, end);
     }
 
     @Nonnull
     protected KeyExpression getSubKeyImpl(int start, int end) {
-        throw new RecordCoreException("Cannot split " + this);
+        throw new UnsplittableKeyExpressionException(this);
```

Comment:
I see. So for a multi-column `FunctionKeyExpression` with the split in the middle, this just throws an error?

I think that's perfectly fine for now.